src/Middleware/Diagnostics/test/UnitTests/ExceptionHandlerTest.cs

```diff
@@ -965,4 +1000,15 @@ public async Task UnhandledError_ExceptionNameTagAdded()
                 Assert.Equal("System.Exception", (string)m.Tags["error.type"]);
             });
     }
+
+    private class TestExceptionHandler : IExceptionHandler
+    {
+        public async ValueTask<bool> TryHandleAsync(HttpContext httpContext, Exception exception, CancellationToken cancellationToken)
+        {
+            httpContext.Response.ContentType = "text/plain";
+            httpContext.Response.StatusCode = 500;
+            await httpContext.Response.WriteAsync($"{nameof(TestExceptionHandler)}: {exception.Message}", cancellationToken: cancellationToken);
+            return true;
```

Comment:
What if TryHandleAsync returns false and there's no handler configured? I don't think the middleware is expecting that.

Reply:
Ah good point! I'm incline to say that in that case we throw an `InvalidOperationException` to be caught and logged here: https://github.com/dotnet/aspnetcore/blob/main/src/Middleware/Diagnostics/src/ExceptionHandler/ExceptionHandlerMiddlewareImpl.cs#L224

Reply:
Sure